### PR TITLE
Add note on special replacement patterns w.r.t. replaceAll

### PR DESCRIPTION
**Pull request by mathiasbynens**

### CODE REVIEW

File: src/features/string-replaceall.md [hunk 1]
```diff
@@ -100,6 +100,22 @@ For consistency with the pre-existing APIs in the language, `String.prototype.re
 
 The important piece of new functionality lies in that first item. `String.prototype.replaceAll` enriches JavaScript with first-class support for global substring replacement, without the need for regular expressions or other workarounds.
 
+## A note on special replacement patterns { #special-patterns }
+
+Worth calling out: both `replace` and `replaceAll` support [special replacement patterns](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_string_as_a_parameter). Although these are most useful in combination with regular expressions, some of them (`$$`, `$&`, ``$` ``, and `$'`) also take effect when performing simple string replacement, which can be surprising:
```
Comment (RReverser) [inline src/features/string-replaceall.md:105]: Woah, TIL!

Comment (mathiasbynens) [reply src/features/string-replaceall.md:105]: FTR, I argued against inheriting this behavior for `replaceAll`, but the committee decided that it'd be confusing for a method named `replaceAll` not to match `replace` in this way, and also that choosing a different name like `substitute` would be confusing. 🤷‍♂️

Comment (LeszekSwirski) [reply src/features/string-replaceall.md:105]: So if you want to replace arbitrary `foo` with arbitrary `bar`, you need to do `string.replaceAll(foo, bar.replaceAll("$","$$$$"))`?

Comment (RReverser) [reply src/features/string-replaceall.md:105]: @LeszekSwirski As far as I understand from the snippet, you can do `string.replaceAll(foo, () => bar)` instead.

Comment (RReverser) [reply src/features/string-replaceall.md:105]: @mathiasbynens I didn't even know that `replace` had that behaviour on strings TBH.

Comment (mathiasbynens) [reply src/features/string-replaceall.md:105]: You're not wrong, Leszek 😭

**Pull request merged**